File: examples_js/estadistica.js
const { sumar, dividir } = require('./calculadora');

function promedio(numeros) {
  if (!numeros || numeros.length === 0) {
    throw new Error('La lista no puede estar vacía');
  }
  const suma = numeros.reduce((acc, n) => sumar(acc, n), 0);
  return dividir(suma, numeros.length);
}

function varianza(numeros) {
  if (!numeros || numeros.length === 0) {
    throw new Error('La lista no puede estar vacía');
  }
  const media = promedio(numeros);
  const suma = numeros.reduce((acc, n) => sumar(acc, (n - media) ** 2), 0);
  return dividir(suma, numeros.length);
}

module.exports = { promedio, varianza };

File: examples_java/Estadistica.java
public class Estadistica {

    private Calculadora calc = new Calculadora();

    public double promedio(int[] numeros) {
        if (numeros == null || numeros.length == 0) {
            throw new IllegalArgumentException("La lista no puede estar vacía");
        }
        int suma = 0;
        for (int n : numeros) {
            suma = calc.suma(suma, n);
        }
        return (double) suma / numeros.length;
    }

    public int productoTotal(int[] numeros) {
        if (numeros == null || numeros.length == 0) {
            throw new IllegalArgumentException("La lista no puede estar vacía");
        }
        int producto = 1;
        for (int n : numeros) {
            producto = calc.multiplicar(producto, n);
        }
        return producto;
    }

    public boolean todosPares(int[] numeros) {
        if (numeros == null || numeros.length == 0) {
            throw new IllegalArgumentException("La lista no puede estar vacía");
        }
        for (int n : numeros) {
            if (!calc.esParejo(n)) {
                return false;
            }
        }
        return true;
    }
}
